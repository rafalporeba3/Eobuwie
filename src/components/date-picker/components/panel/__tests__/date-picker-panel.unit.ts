import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils';
import CompositionApi                            from '@vue/composition-api';
import cDatePickerPanel                          from '../date-picker-panel.component.vue';

const localVue = createLocalVue();
localVue.use(CompositionApi);

describe('Date picker panel component', () => {
  let wrapper!: Wrapper<any>;

  beforeEach(() => {
    wrapper = shallowMount(cDatePickerPanel, {
      localVue,
      slots: {
        header: '<div></div>',
        content: '<div></div>',
      },
    });
  });

  it('should render cDatePickerPanel component with correct class', () => {
    expect(wrapper.classes())
      .toContain('c-date-picker-panel');
  });

  it('should render slot wrapper when slot has been passed', () => {
    expect(wrapper.find('.c-date-picker-panel__header')
      .exists())
      .toBeTruthy();

    expect(wrapper.find('.c-date-picker-panel__content')
      .exists())
      .toBeTruthy();
  });

  it('should not render slot wrapper when slot is empty', () => {
    wrapper = shallowMount(cDatePickerPanel,
      {
        localVue,
        slots: {},
      });

    const header: Wrapper<any> = wrapper.find('.c-date-picker-panel__header');
    const content: Wrapper<any> = wrapper.find('.c-date-picker-panel__content');

    expect(header.exists())
      .toBeFalsy();
    expect(content.exists())
      .toBeFalsy();
  });

  it('should match snapshot', () => {
    expect(wrapper.html())
      .toMatchSnapshot();
  });
});
