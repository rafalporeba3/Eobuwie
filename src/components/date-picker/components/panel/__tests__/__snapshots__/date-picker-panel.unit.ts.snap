// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date picker panel component should match snapshot 1`] = `
<div class="c-date-picker-panel">
  <div class="c-date-picker-panel__header">
    <div></div>
  </div>
  <div class="c-date-picker-panel__content">
    <div></div>
  </div>
</div>
`;
